go/src/golangApi/main/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+  "database/sql"
+  "fmt"
+  "log"
+  "net/http"
+  "encoding/json"
+  d "golangApi/db"
+  o "golangApi/models"
+
+  _ "github.com/lib/pq"
+)
+
+var db *sql.DB
+
+func main() {
+  d.InitDb()
+  fmt.Println("Database connection established")
+  defer db.Close()
+  http.HandleFunc("/", helloWorldHandler)
+  http.HandleFunc("/api/v1/olympians/", olympiansHandler)
+  if err := http.ListenAndServe("localhost:8080", nil); err != nil {
+    log.Fatal("ListenAndServe: ", err)
+}
+}
+
+func helloWorldHandler(w http.ResponseWriter, req *http.Request) {
```

Comment:
Is the *http.Request defining what type `req` must be?

Reply:
Handler functions come with the http package, and they require ResponseWriter and Request as arguments. In this case I'm not really using the req for anything, just have it in there because it's a required argument.

Reply:
it would allow me to get request headers, params, etc if I needed it.